ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle, buildStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface FolderPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+function styles_(props:FolderPropertiesDialogProps) {
+	return buildStyle('FolderPropertiesDialog', props.theme, (theme: any) => {
+		return {
+			folderLabel: {
+				...theme.textStyle,
+				flex: 1,
+				display: 'flex',
+				fontWeight: 'bold',
+				color: theme.color,
+			},
+			folderSaveButton: {
+				background: 'none',
+				border: 'none',
+			},
+			folderSaveButtonIcon: {
+				color: theme.color,
+				fontSize: '1.4em',
+			},
+			titleForm: {
+				display: 'inline-block',
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+				border: '1px solid',
+				borderColor: theme.dividerColor,
+				marginLeft: '3em',
+				width: '6em',
+			},
+			iconForm: {
+				display: 'inline-block',
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+				border: '1px solid',
+				borderColor: theme.dividerColor,
+				marginLeft: '3.6em',
+				width: '6em',
+			},
+		};
+	});
+}
+
+export default function FolderPropertiesDialog(props: FolderPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [formFolder, setFormFolder] = useState<any>({});
+
+	const theme = themeStyle(props.theme);
+	const styles = styles_(props);
+
+	useEffect(() => {
+		async function fetchFolder() {
+			if (!props.folderId) {
+				throw new Error('FolderID is not set. This Notebook is not valid!!!');
```

Comment:
What happens when an error is thrown? Is it caught somewhere? Does it crash the application?

Reply:
So are all errors thrown being caught by Electron bridge or is there a specific set for them?